services/github/github-workflow-status.service.js

```diff
@@ -11,6 +11,10 @@ const schema = Joi.object({
     .required(),
 }).required()
 
+const queryParamSchema = Joi.object({
+  event: Joi.string(),
```

Comment:
Was wondering if it would be worth a more restrictive schema that had a list of allowed events, though seems like there's quite a few of them and probably not worth it

Reply:
My personal opinion is that some of the events are not particularly interesting for a project like shields, so restricting the user input doesn't really make sense and is unnecessary. If the user sets an invalid event name, they will see the "no status" response anyway. But if that's how it's handled in other services with a set of optional parameters, then I could of course add a list of the currently supported events by Github to the schema. If Github changes the events list in the future, you'll have to upgrade this though, adding maintenance burden.
https://docs.github.com/en/actions/reference/events-that-trigger-workflows

Reply:
That's always the tradeoff. In this case the tighter restriction mainly just helps users spot typos more easily vs. other cases where it ensures that only values the service class can deal with pass the filter. 

Given the long list I think it's better to go without it, can always reevaluate later